base/iterators.jl

```diff
@@ -1162,6 +1162,53 @@ end
 reverse(f::Flatten) = Flatten(reverse(itr) for itr in reverse(f.it))
 last(f::Flatten) = last(last(f.it))
 
+"""
+    Iterators.flatmap(f, iterators...)
+
+Equivalent to flatten(map(f, iterators...)).
+
+# Examples
+```jldoctest
+julia> flatmap(n->-n:2:n, 1:3) |> collect
+9-element Vector{Int64}:
+ -1
+  1
+ -2
+  0
+  2
+ -3
+ -1
+  1
+  3
+
+julia> flatmap(x -> (x+1)*x % 4 == 0 ? (x*x,) : (), 1:11) |> collect
+5-element Vector{Int64}:
+   9
+  16
+  49
+  64
+ 121
+
+julia> [(j,k) for j in 1:3 for k in 1:3 if j>k]
```

Comment:
`````suggestion
```

(Write some explanation here to mention that nested comprehension is
expressed as flatten and also that a branching can be expressed as a
Union of 0-tuple and 1-tuple?)

```jldoctest
julia> [(j,k) for j in 1:3 for k in 1:3 if j>k]
`````

Reply:
To be honest, I tried to be purposefully concise here and just let the code speak. "Here's a two-level comprehension with a filter, here's the same result using flatmap". Also a bit like a puzzle. I leave it up to you to add any more insights if you think it's worth it.